src/main/java/org/opensearch/geospatial/ip2geo/action/UpdateDatasourceRequest.java

```diff
@@ -0,0 +1,168 @@
+    }
+
+    @Override
+    public void writeTo(final StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(name);
+        out.writeOptionalString(endpoint);
+        out.writeOptionalTimeValue(updateInterval);
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException errors = new ActionRequestValidationException();
+        if (endpoint == null && updateInterval == null) {
+            errors.addValidationError("no values to update");
+        }
+
+        validateEndpoint(errors);
+        validateUpdateInterval(errors);
+
+        return errors.validationErrors().isEmpty() ? null : errors;
+    }
+
+    /**
+     * Conduct following validation on endpoint
+     * 1. endpoint format complies with RFC-2396
+     * 2. validate manifest file from the endpoint
+     *
+     * @param errors the errors to add error messages
+     */
+    private void validateEndpoint(final ActionRequestValidationException errors) {
+        if (endpoint == null) {
+            return;
+        }
+
+        try {
+            URL url = new URL(endpoint);
+            url.toURI(); // Validate URL complies with RFC-2396
+            validateManifestFile(url, errors);
+        } catch (MalformedURLException | URISyntaxException e) {
+            log.info("Invalid URL[{}] is provided", endpoint, e);
+            errors.addValidationError("Invalid URL format is provided");
+        }
+    }
+
+    /**
+     * Conduct following validation on url
+     * 1. can read manifest file from the endpoint
+     * 2. the url in the manifest file complies with RFC-2396
+     *
+     * @param url the url to validate
+     * @param errors the errors to add error messages
+     */
+    private void validateManifestFile(final URL url, final ActionRequestValidationException errors) {
+        DatasourceManifest manifest;
+        try {
+            manifest = DatasourceManifest.Builder.build(url);
+        } catch (Exception e) {
+            log.info("Error occurred while reading a file from {}", url, e);
+            errors.addValidationError(String.format(Locale.ROOT, "Error occurred while reading a file from %s: %s", url, e.getMessage()));
+            return;
+        }
+
+        try {
+            new URL(manifest.getUrl()).toURI(); // Validate URL complies with RFC-2396
+        } catch (MalformedURLException | URISyntaxException e) {
+            log.info("Invalid URL[{}] is provided for url field in the manifest file", manifest.getUrl(), e);
+            errors.addValidationError("Invalid URL format is provided for url field in the manifest file");
+        }
+    }
+
+    /**
+     * Validate updateInterval is equal or larger than 1
+     *
+     * @param errors the errors to add error messages
+     */
+    private void validateUpdateInterval(final ActionRequestValidationException errors) {
+        if (updateInterval == null) {
+            return;
```

Comment:
is null update interval valid?

Reply:
If the value is null, we don't update the value. As long as one of values are not null, the request is valid.